docs/drafts/derive-serviceaccount.md

```diff
@@ -0,0 +1,578 @@
+                - delete
+                - get
+                - list
+                - patch
+                - update
+                - watch
+```
+
+Below is the translation of the above cluster permissions into ClusterRole and ClusterRoleBinding for your cluster extension:
+
+```yml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: <cluster-extension-name>
+rules:
+- apiGroups:
+  - olm.operatorframework.io
+  resourceNames:
+  - <cluster-extension-name>
+  resources:
+  - clusterextensions/finalizers
+  verbs:
+  - update
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - clusterroles
+  - clusterrolebindings
+  verbs:
+  - create
+- apiGroups:
+  - user.openshift.io
+  resources:
+  - users
+  - groups
+  - identities
+  verbs:
+  - get
+  - list
+```
+
+Below is the ClusterRoleBinding that associates your cluster extension to its service account:
+
+```yml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: <cluster-extension-name>
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: <cluster-extension-name>
+subjects:
+- kind: ServiceAccount
+  name: <cluster-extension-sa-name>
+  namespace: <cluster-extension-namespace>
+```
+
+Below is the translation of the above cluster permissions into Role and RoleBinding for your cluster extension:
+
+```yml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: <cluster-extension-name>
+  namespace: <cluster-extension-name>
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  - services
+  - services/finalizers
+  - endpoints
+  - persistentvolumeclaims
+  - events
+  - configmaps
+  - secrets
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - apps
+  resourceNames:
+  - <cluster-extension-name>
+  resources:
+  - deployments/finalizers
+  verbs:
+  - update
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - get
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - serviceaccounts
+  verbs:
+  - create
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - roles
+  - rolebindings
+  verbs:
+  - create
+```
+Below is the RoleBinding that associates your cluster extension to its service account:
+
+```yml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: <cluster-extension-name>
+  namespace: <cluster-extension-namespace>
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: <cluster-extension-name>
+subjects:
+- kind: ServiceAccount
+  name: <cluster-extension-sa-name>
+```
+
+You can use the above example as a starting point to derive the minimal RBAC for your cluster extension.
```

Comment:
Following on from my suggestion at the top of the section, I think each of these points should be illustrated showing where the the input came from (e.g. given that there is a Service in the bundle, we need to add a service permissions over here...)